indexer/indexer_tests/osm_editor_test.cpp

```diff
@@ -0,0 +1,558 @@
+  });
+  ASSERT(mwmId.GetInfo(), ());
+
+  ForEachCafeAtPoint(m_index, m2::PointD(1.0, 1.0), [&editor, &mwmId](FeatureType & ft)
+  {
+    TEST(!editor.GetFeatureTypeInfo(mwmId, ft.GetID().m_index), ());
+
+    osm::EditableMapObject emo;
+    FillEditableMapObject(editor, ft, emo);
+    emo.SetBuildingLevels("1");
+    TEST_EQUAL(editor.SaveEditedFeature(emo), osm::Editor::SaveResult::SavedSuccessfully, ());
+
+    auto const fti = editor.GetFeatureTypeInfo(mwmId, ft.GetID().m_index);
+    TEST_EQUAL(fti->m_feature.GetID(), ft.GetID(), ());
+  });
+}
+
+void EditorTest::GetEditedFeatureTest()
+{
+  auto & editor = osm::Editor::Instance();
+
+  {
+    FeatureID feature;
+    FeatureType ft;
+    TEST(!editor.GetEditedFeature(feature, ft), ());
+  }
+
+  TestCafe cafe(m2::PointD(1.0, 1.0), "London Cafe", "en");
+  auto const mwmId = ConstructTestMwm([&cafe](TestMwmBuilder & builder)
+  {
+    builder.Add(cafe);
+  });
+  ASSERT(mwmId.GetInfo(), ());
+
+  ForEachCafeAtPoint(m_index, m2::PointD(1.0, 1.0), [&editor](FeatureType & ft)
+  {
+    FeatureType featureType;
+    TEST(!editor.GetEditedFeature(ft.GetID(), featureType), ());
+
+    osm::EditableMapObject emo;
+    FillEditableMapObject(editor, ft, emo);
+    emo.SetBuildingLevels("1");
+    TEST_EQUAL(editor.SaveEditedFeature(emo), osm::Editor::SaveResult::SavedSuccessfully, ());
+
+    TEST(editor.GetEditedFeature(ft.GetID(), featureType), ());
+    TEST_EQUAL(ft.GetID(), featureType.GetID(), ());
+  });
+}
+
+void EditorTest::GetEditedFeatureStreetTest()
+{
+  auto & editor = osm::Editor::Instance();
+
+  {
+    FeatureID feature;
+    string street;
+    TEST(!editor.GetEditedFeatureStreet(feature, street), ());
+  }
+
+  TestCafe cafe(m2::PointD(1.0, 1.0), "London Cafe", "en");
+  auto const mwmId = ConstructTestMwm([&cafe](TestMwmBuilder & builder)
+  {
+    builder.Add(cafe);
+  });
+  ASSERT(mwmId.GetInfo(), ());
+
+  ForEachCafeAtPoint(m_index, m2::PointD(1.0, 1.0), [&editor](FeatureType & ft)
+  {
+    string street;
+    TEST(!editor.GetEditedFeatureStreet(ft.GetID(), street), ());
+
+    osm::EditableMapObject emo;
+    FillEditableMapObject(editor, ft, emo);
+    osm::LocalizedStreet ls{"some street", ""};
+    emo.SetStreet(ls);
+    TEST_EQUAL(editor.SaveEditedFeature(emo), osm::Editor::SaveResult::SavedSuccessfully, ());
+
+    TEST(editor.GetEditedFeatureStreet(ft.GetID(), street), ());
+    TEST_EQUAL(street, ls.m_defaultName, ());
+  });
+}
+
+void EditorTest::OriginalFeatureHasDefaultNameTest()
+{
+  auto & editor = osm::Editor::Instance();
+
+  TestCafe cafe(m2::PointD(1.0, 1.0), "London Cafe", "en");
+  TestCafe unnamedCafe(m2::PointD(2.0, 2.0), "", "en");
+  TestCafe secondUnnamedCafe(m2::PointD(3.0, 3.0), "", "en");
+
+  auto const mwmId = ConstructTestMwm([&cafe, &unnamedCafe, &secondUnnamedCafe](TestMwmBuilder & builder)
+  {
+    builder.Add(cafe);
+    builder.Add(unnamedCafe);
+    builder.Add(secondUnnamedCafe);
+  });
+  ASSERT(mwmId.GetInfo(), ());
```

Comment:
Please, don't call GetInfo() unless you're planning to use MwmInfo. Use IsAlive() instead.